lib/src/shared/main/java/com/couchbase/lite/Blob.java

```diff
@@ -296,13 +296,14 @@ public Blob(@NonNull String contentType, @NonNull URL fileURL) throws IOExceptio
         }
 
         this.contentType = contentType;
-        this.initialContentStream = fileURL.openStream();
-        this.blobLength = 0L; // unknown
+
+        initStream(fileURL.openStream());
```

Comment:
Shall we make sure to close the opened stream in finalize?

Reply:
Yeah.  Java's base stream class already does exactly that.  It is kind of lame, because it is possible that it will take hours to do that, but we really can't do much better...

Reply:
I don't know Java already does that by default.

Reply:
NVM.. I'm going to add a finalizer.  There *are* streams that don't close themselves...